src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -1,135 +1,252 @@
-			};
-			$scope.sortInfo = {
-				fields: [], directions: [], columns: []
+				gridContainer.style.height = (window.innerHeight - gridContainer.offsetTop - footer.offsetHeight) + "px";
 			};
-			$scope.getBookRange = function (count, currentPage, searchText) {
-				var first = (currentPage - 1) * count;
-				var sortField = $scope.sortInfo.fields[0];
-				var sortBy = null;
-				// Todo: setting sortBy to a complex field like this does not work...no sorting happens.
-				// It appears we will need to put a redundant top-level data field in the record for
-				// anything we want to sort by. This is especially annoying in that it may not actually
-				// be useful to sort by all the fields...but ng-grid allows the user to do so.
-				// Possibly we could disallow some fields with a message and switch to some other sort.
-				// Another reason a redundant field is probably necessary is that parse.com's built-in sorting
-				// is case-sensitive, and their only suggestion for overcoming this is a redundant
-				// all-lower-case field. Similarly any other sort-key field must somehow be in a form
-				// where a plain binary sort will give the right results.
-				if (sortField == 'title') {
-					sortBy = 'volumeInfo.title';
-				}
-				bookService.getFilteredBookRange(first, count, searchText, "", "", "", sortBy).then(function (result) {
-					$scope.visibleBooks = result;
-					$scope.visibleData = $scope.visibleBooks.map(function (item) {
-						return {
-							title: item.volumeInfo.title,
-							published: item.volumeInfo.publishedDate,
-							publisher: item.volumeInfo.publisher,
-							pages: item.volumeInfo.pageCount
-						};
+
+			window.addEventListener("resize", resizeGrid);
+
+			$scope.getBooks = function() {
+				var first = 0;
+				bookService.getFilteredBooksCount('', '', '', '', true).then(function (result) {
+					var count = result;
+
+					bookService.getFilteredBookRange(first, count, '', '', '', '', '', '', true).then(function (result) {
+						$scope.booksData = result.map(function (item) {
+							return {
+								//Hidden id
+								objectId: item.objectId,
+								inCirculation: item.inCirculation !== false ? 'yes' : 'no',
+								title: item.title,
+								createdAt: (function () {
+									var dateWithTime = new Date(item.createdAt);
+									return new Date(dateWithTime.getFullYear(), dateWithTime.getMonth(), dateWithTime.getDate());
+								}()),
+								copyright: item.copyright.match("^Copyright ") ? item.copyright.substring(10) : item.copyright,
+								license: item.license,
+								updatedAt: (function () {
+									var dateWithTime = new Date(item.updatedAt);
+									return new Date(dateWithTime.getFullYear(), dateWithTime.getMonth(), dateWithTime.getDate());
+								}()),
+								pageCount: item.pageCount,
+								bookshelf: item.bookshelf,
+								tags: item.tags ? item.tags.map(function(item) {
+									//Put into tagList; Todo: remove when tagService is updated
```

Comment:
I had put those notes in expecting us to switch to server-side paging in the near future. Since we are still not switching to that, I have left that code commented out with notes on the condition for using.